src/CVQualifiers.cpp

```diff
@@ -298,22 +318,42 @@ CVQualifiers::random_qualifiers(const Type* t, Effect::Access access, const CGCo
 	const Effect &effect_context = cg_context.get_effect_context();
 
 	// set random volatile/const properties for each level of indirection for pointers
+	// First set up the vectors with correct number of qualifiers:
+	unsigned level = 0;
 	const Type* tmp = t->ptr_type;
 	while (tmp) {
-		DEPTH_GUARD_BY_DEPTH_RETURN(2, ret_qfer);
-		isVolatile = rnd_flipcoin(volatile_prob);
-		ERROR_GUARD(ret_qfer);
+		++level;
+		is_consts.push_back(false);
+		is_volatiles.push_back(false);
+		tmp = tmp->ptr_type;
+	}
+	// Then make the random properties (properties need to be in reverse order): 
+	tmp = t->ptr_type;
+	while (tmp) {   
+		bool volatile_ok = is_volatile_ok_on_one_level(tmp);
+		if (volatile_ok){
+			DEPTH_GUARD_BY_DEPTH_RETURN(2, ret_qfer);
```

Comment:
It's fine to remove DEPTH_GUARD_BY_DEPTH_RETURN, which was introduced for bounded-depth testing. It is not used any more. We should've removed it long time ago.